app/questionnaire/dependencies.py

```diff
@@ -20,7 +20,7 @@ def get_routing_path_block_ids_by_section_for_dependent_sections(
     location: LocationType,
     progress_store: ProgressStore,
     path_finder: PathFinder,
-    source_type: str,
+    source_types: set[str],
```

Comment:
Why did you decide to use set here? There are arguments, like `sections_to_ignore` that are passed as a list and not modified in the method. maybe we should keep it consistent and use sets or lists for all arguments of this method that it applies to? I mean, there is no difference in how `sections_to_ignore` and `source_types` are used later?

Reply:
Yeah set is not needed, it can be more generic so I've changed it to Iterable, unlike sections to ignore theres no case of `source_types or []` or similar, so I think a totally generic type is probably what this should be